context.c

```diff
@@ -596,11 +596,10 @@ iio_create_context_from_xml(const struct iio_context_params *params,
 			prm_err(params, "Unable to alloc memory\n");
 			goto err_context_destroy;
 		}
-
-		free(ctx->description);
-		ctx->description = new_description;
 	}
 
+	free(ctx->description);
+	ctx->description = new_description;
```

Comment:
If `description == null` and `ctx->description` is a valid pointer this will free the valid pointer and then assign `new_description` (which is `null`) to `ctx-description`.

Reply:
correct... good catch!

Reply:
Well, I disagree. If `description` is NULL it makes sense that `ctx->description` is set to NULL, no?

Reply:
Nevermind, I think you are right. It *would* make sense if the description was replaced, but the code tries to append the old description to the new one.
So please send a PR.

Reply:
Thanks for the quick replies! Just out of curiosity, how is `description` here normally populated? In my case I'm running `iio_info -u "serial:..."` and IIOD on an MCU.

Reply:
Here it is populated by the XML backend from the XML string sent by IIOD, which itself will get the description from the libiio backend's `iio_backend` struct.

Reply:
Sorry that wasn't phrased clearly. :sweat_smile:  I understand how it works in my case.

I was curious in what cases the `description` that is passed in as an argument to `iio_create_context_from_xml()` will not be `null`, which I assume is the "normal" use case.

Reply:
It happens e.g. with the network backend, which prepends the IP address to the description found in the XML string.